### ISSUE TITLE
Dependency compilation error when clustering is enabled

### ISSUE THREAD

**Issue by axelson** [sasa1977/boundary#28]
I am getting a rather odd error that I think is a bug in boundary. I have two applications, the sub application is a path dependency with `runtime: false`. I get the error after I run mix compile, and then run with clustering enabled `iex --sname tmp -S mix`.

This results in the following error:
```
$ iex --sname tmp -S mix
Erlang/OTP 23 [erts-11.1.2] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [hipe]

==> sub_app
could not compile dependency :sub_app, "mix compile" failed. You can recompile this dependency with "mix deps.compile sub_app", update it with "mix deps.update sub_app" or clean it with "mix deps.clean sub_app"
** (ArgumentError) argument error
    :erlang.is_process_alive(#PID<3.278.0>)
    (elixir 1.11.2) lib/code.ex:700: Code.eval_quoted/3
    (boundary 0.7.0) lib/boundary/definition.ex:96: anonymous fn/2 in Boundary.Definition.decode/1
    (elixir 1.11.2) lib/map.ex:818: Map.update!/3
    (boundary 0.7.0) lib/boundary/definition.ex:45: Boundary.Definition.get/1
    (boundary 0.7.0) lib/boundary/view.ex:94: anonymous fn/4 in Boundary.View.load_app_boundaries/3
```

Here is a reproduction repository: https://github.com/axelson/boundary_cluster_error_repro

**Comment by sasa1977** [comment]
Thanks for the report. I've pushed a potential fix in [this branch](https://github.com/sasa1977/boundary/tree/fix-node-crash). Would you mind giving it a try before I merge?

**Comment by axelson** [comment]
Okay, I'm no longer getting the crash on the repro project and my actual project. Thanks!

**Comment by sasa1977** [comment]
FYI, this is now published on hex in 0.7.1

**Comment by axelson** [comment]
Great, thanks!
